### PR TITLE
Add PreferAny option to CPU preferences

### PR DESCRIPTION
**Pull request by opokornyy**
**What this PR does / why we need it**:

This commit adds PreferAny option to CPU preferences that should allow a user to accept any topology when there are required number of vCPUs provided.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```
None
```

### CODE REVIEW

File: pkg/instancetype/instancetype_test.go [hunk 1]
```diff
@@ -2060,6 +2072,28 @@ var _ = Describe("Instancetype and Preferences", func() {
 					},
 				},
 			),
+			Entry("by a VM for vCPUs using PreferAny",
```
Comment (lyarwood) [inline pkg/instancetype/instancetype_test.go:2075]: Can you add coverage when an instance type is also used?

File: staging/src/kubevirt.io/api/instancetype/v1beta1/types.go [hunk 2]
```diff
@@ -315,6 +315,11 @@ const (
 
 	// Prefer vCPUs to be spread evenly between cores and sockets with any remaining vCPUs being presented as cores
 	PreferSpread PreferredCPUTopology = "preferSpread"
+
+	// Prefer vCPUs to be spread according to VirtualMachineInstanceTemplateSpec
+	//
+	// If used with VirtualMachineInstance it will use sockets as default
```
Comment (lyarwood) [inline staging/src/kubevirt.io/api/instancetype/v1beta1/types.go:321]: `VirtualMachineInstancetype`

File: pkg/instancetype/instancetype.go [hunk 3]
```diff
@@ -964,7 +969,7 @@ func applyCPU(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.Virtua
 	switch GetPreferredTopology(preferenceSpec) {
 	case instancetypev1beta1.PreferCores:
 		vmiSpec.Domain.CPU.Cores = instancetypeSpec.CPU.Guest
-	case instancetypev1beta1.PreferSockets:
+	case instancetypev1beta1.PreferSockets, instancetypev1beta1.PreferAny:
```
Comment (lyarwood) [inline pkg/instancetype/instancetype.go:972]: Need test coverage for this.

**Review by lyarwood — COMMENTED** [review]
/area instancetype
/cc

Looks like a good start to me, some missing test coverage but I think this is the correct direction.

File: pkg/instancetype/instancetype.go (hunk 3 above)
Comment (opokornyy) [reply pkg/instancetype/instancetype.go:972]: This should be covered in this test - https://github.com/kubevirt/kubevirt/pull/10547/files#diff-218f2952941d0ec349e5274a0cc9d5399f23c596afad577f824d323b81cd7422R1033

**Review by 0xFelix — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**